src/project-manager.ts

```diff
@@ -951,6 +971,28 @@ export class ProjectConfiguration {
 	private ensuredAllFiles = false;
 
 	/**
+	 * Ensures a single file is available to the LanguageServiceHost
+	 * @param filePath
+	 */
+	ensureSourceFile(filePath: string): boolean {
```

Comment:
As I understand it, the return type indicates whether the program was resynced as part of the method call? Why exactly is that information needed for a caller who wants to ensure a source file? If it is needed to decide whether to republish diagnostics, then this puts unnecessary burden on the callers to always check the return value and act on it - emitting an event is better in that case